static/src/js/util/accessMethods/buildAccessMethods/__tests__/buildHarmony.test.js

```diff
@@ -0,0 +1,761 @@
+          nativeId: 'mmt_variable_3970',
+          scienceKeywords: null
+        }, {
+          conceptId: 'V100003-EDSC',
+          definition: 'Red channel value',
+          longName: 'Red Channel',
+          name: 'red_var',
+          nativeId: 'mmt_variable_3969',
+          scienceKeywords: null
+        }]
+      }
+    }
+
+    const { services, variables: associatedVariables } = collectionMetadata
+    const serviceItem = services.items[0]
+
+    const index = 0
+
+    const params = {
+      associatedVariables,
+      index
+    }
+
+    const methods = buildHarmony(serviceItem, params)
+
+    expect(methods).toEqual({
+      harmony0: {
+        description: 'abc123',
+        enableTemporalSubsetting: true,
+        enableSpatialSubsetting: true,
+        hierarchyMappings: [
+          {
+            id: 'V100000-EDSC'
+          },
+          {
+            id: 'V100001-EDSC'
+          },
+          {
+            id: 'V100002-EDSC'
+          },
+          {
+            id: 'V100003-EDSC'
+          }
+        ],
+        id: 'S100000-EDSC',
+        isValid: true,
+        keywordMappings: [],
+        longName: 'Mock Service Name',
+        name: 'mock-name',
+        supportedOutputFormats: [
+          'GEOTIFF',
+          'PNG',
+          'TIFF',
+          'NETCDF-4'
+        ],
+        supportedOutputProjections: [],
+        supportsBoundingBoxSubsetting: true,
+        supportsShapefileSubsetting: false,
+        supportsTemporalSubsetting: false,
+        supportsVariableSubsetting: true,
+        supportsConcatenation: true,
+        defaultConcatenation: true,
+        enableConcatenateDownload: true,
+        type: 'Harmony',
+        url: 'https://example.com',
+        variables: {
+          'V100000-EDSC': {
+            conceptId: 'V100000-EDSC',
+            definition: 'Alpha channel value',
+            longName: 'Alpha channel ',
+            name: 'alpha_var',
+            nativeId: 'mmt_variable_3972',
+            scienceKeywords: null
+          },
+          'V100001-EDSC': {
+            conceptId: 'V100001-EDSC',
+            definition: 'Blue channel value',
+            longName: 'Blue channel',
+            name: 'blue_var',
+            nativeId: 'mmt_variable_3971',
+            scienceKeywords: null
+          },
+          'V100002-EDSC': {
+            conceptId: 'V100002-EDSC',
+            definition: 'Green channel value',
+            longName: 'Green channel',
+            name: 'green_var',
+            nativeId: 'mmt_variable_3970',
+            scienceKeywords: null
+          },
+          'V100003-EDSC': {
+            conceptId: 'V100003-EDSC',
+            definition: 'Red channel value',
+            longName: 'Red Channel',
+            name: 'red_var',
+            nativeId: 'mmt_variable_3969',
+            scienceKeywords: null
+          }
+        }
+      }
+    })
+  })
+
+  test('increments index correctly for multiple harmony items', () => {
```

Comment:
Nothing in `buildHarmony` increments an index. This test is duplicating functionality written in `buildAccessMethods`, but not actually testing anything. 

You should have tests in that file that ensure the indexes are incremented correctly for all 3 index types